android/src/main/java/com/rudderstack/android/AnalyticsRegistry.kt

```diff
@@ -0,0 +1,54 @@
+package com.rudderstack.android
+
+import androidx.annotation.VisibleForTesting
+import com.rudderstack.core.Analytics
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * AnalyticsRegistry is a singleton object responsible for managing and providing instances of the Analytics class.
+ * It maintains a mapping between write keys and Analytics instances using concurrent hash maps.
+ *
+ * The class provides methods for registering new Analytics instances with unique write keys,
+ * as well as retrieving instances based on write keys.
+ *
+ * Usage:
+ * - To register a new Analytics instance, use the [register] method, providing a write key and the Analytics instance.
+ * - To retrieve an Analytics instance, use the [getInstance] method, passing the write key.
+ *
+ * Note: The class is marked as internal, indicating that it is intended for use within the same module and should not be accessed
+ * from outside the module.
+ */
+internal object AnalyticsRegistry {
```

Comment:
Why do we need an extra object? the methods could be inside the file only.

Reply:
I've kept an extra object to encapsulate the logic related to multiple instances handling.

Which file you're referring to over here? Is it the `RudderAnalytics.kt` class?

Just to clarify: Are you suggesting moving the `AnalyticsRegistry` methods to some other place?

Reply:
Remove the line "internal object AnalyticsRegistry {". Kotlin already creates a class based on the file name. Moreover these methods can be static, so annotate them with @JvmStatic.

Reply:
When I'm trying to add `@JvmStatic` to methods, after removing the said line, I'm getting this error:
> Only members in named objects and companion objects can be annotated with '@JvmStatic'
Please LMK what should I do?